tripal/src/TripalImporter/TripalImporterBase.php

```diff
@@ -149,14 +216,33 @@ public function __construct(array $configuration, $plugin_id, $plugin_definition
     $this->prev_update = 0;
     $this->reported = 0;
 
-    // Initialize the logger.
-    $this->logger = \Drupal::service('tripal.logger');
+    // Initialize the Drupal messenger
+    if ($messenger === NULL) {
+      @trigger_error('Calling ' . __METHOD__ . '() without the $messenger argument is deprecated in tripal 4.0.0 and it will be required in tripal:4.1.0. To resolve this, make sure the create() method in your importer grabs the Drupal\Core\Messenger\Messenger from the container and supplies it to the parent in the constructor.', E_USER_DEPRECATED);
+      $messenger = \Drupal::messenger();
+    }
+    $this->messenger = $messenger;
 
-    // Initialize messenger
-    $this->messenger = \Drupal::messenger();
+    // Initialize the Tripal logger.
+    if ($logger === NULL) {
+      @trigger_error('Calling ' . __METHOD__ . '() without the $logger argument is deprecated in tripal 4.0.0 and it will be required in tripal:4.1.0. To resolve this, make sure the create() method in your importer grabs the Drupal\tripal\Services\TripalLogger from the container and supplies it to the parent in the constructor.', E_USER_DEPRECATED);
```

Comment:
```suggestion
      @trigger_error('Calling ' . __METHOD__ . '() without the $logger argument is deprecated in tripal 4.0.0-alpha3 and it will be required in tripal:4.1.0. To resolve this, make sure the create() method in your importer grabs the Drupal\tripal\Services\TripalLogger from the container and supplies it to the parent in the constructor.', E_USER_DEPRECATED);
```